src/NationalInstruments.Analyzers/Correctness/ApprovedNamespaceCodeFixProvider.cs

```diff
@@ -84,6 +84,10 @@ protected override Task<Document> GetChangedDocumentAsync(CancellationToken canc
                 return Task.FromResult(_context.Document);
             }
 
+            [SuppressMessage(
```

Comment:
Is there an easy fix to address this warning rather than suppressing it?

Reply:
I'm not exactly sure, it seems there's no way around writing the approved namespace file.

Also, I think the warning is overly aggressive because this is a code fix provider, not an analyzer—I think it shouldn't even participate in compilation?  The whole point of banning file IO during compilation is because it is tricky during incremental build.  Code fix providers shouldn't affect building (as far as I understand it).  

Separately, the warning that this code fix provider is fixing, NI1800, is not enabled by default.  In ASW we also disable it with this comment:

```ini
# Missing the namespace approval file - we don't do this namespace analysis with analyzers yet.
dotnet_diagnostic.NI1800.severity = none
```

So it seems this rule and code fix provider isn't really being used currently.

Reply:
Okay. @pakdev are you okay with this warning in this analyzer? Is this indicative of another issue that we need to resolve "soon" ?

Reply:
Not worried about this for the reasons @jonathanou stated